Added Several Tags and Mechanisms

BehrRiley opened the pull request:
Adds 7 simple boolean tags and mechanisms. Rebased from my last broken PR

BehrRiley committed:
Added Several Tags and Mechanisms

mcmonkey4eva commented on plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialCommandBlock.java:
```diff
@@ -0,0 +1,83 @@
+import com.denizenscript.denizencore.objects.properties.PropertyParser;
+import org.bukkit.block.data.type.CommandBlock;
+
+public class MaterialCommandBlock implements Property {
```
class name doesn't match property

mcmonkey4eva replied:
Same problem on all other classes PR'd as well

mcmonkey4eva commented on plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialGate.java:
```diff
@@ -0,0 +1,83 @@
+        // @mechanism MaterialTag.signal_fire
+        // @group properties
+        // @description
+        // Returns whether this campfire will produce longer smoke trails, or not.
```
This is just outright wrong copypastad meta

mcmonkey4eva commented on plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialCommandBlock.java:
```diff
@@ -0,0 +1,83 @@
+        // @mechanism MaterialTag.conditional
+        // @group properties
+        // @description
+        // Returns whether this command block is conditional, or not.
```
Many of these seem like they'd easily fit into `MaterialMode`.

BehrRiley replied:
I'll merge this into `MaterialMode` in another set of changes I make later.

BehrRiley committed:
Resolved Reviews

- Renamed classes from the block data type name to property name
- fixed bad copypasta typo
- removed CommandBlock property, merging with Mode property later
- renamed & alphabetized Property registry

mcmonkey4eva commented:
`Resolved Reviews`
Pressing the "Resolve" button doesn't actually resolve them (in the same way that minimizing your console doesn't fix errors).

BehrRiley commented:
Closing for the moment - will PR again in the future when I have a bit more time to work on it.
